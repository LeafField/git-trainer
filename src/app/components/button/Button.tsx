"use client";
import React, { FC } from "react";
import Link from "next/link";

type Props = {
  title: string;
  link: string;
  color?: "primary" | "secondary";
};

const Button: FC<Props> = ({ title, link, color = "primary" }) => {
  return (
    <Link
      className={`block w-fit rounded-[2.8125rem] px-5 py-2 text-base font-normal text-white after:block after:h-[1px]
      after:w-full after:scale-0 after:transition-all hover:after:scale-100 hover:after:bg-white
      hover:after:content-[""] md:px-10 md:py-4 md:text-2xl
      ${color === "primary" && "bg-primary"} ${
        color === "secondary" && "border border-white"
      }`}
      href={`${link}`}
    >
      {title}
    </Link>
  );
};

export default Button;
