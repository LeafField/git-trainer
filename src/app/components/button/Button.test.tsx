import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Buttonのレンダリングテスト", () => {
  test("primaryのレンダリングテスト", () => {
    const { container } = render(
      <Button color="primary" link="/difficulty" title="Let's Play!" />,
    );
    expect(
      screen.getByRole("link", { name: "Let's Play!" }),
    ).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveAttribute("href", "/difficulty");
    expect(container).toMatchSnapshot();
  });

  test("secondaryのレンダリングテスト", () => {
    const { container } = render(
      <Button link="/howto" title="How to use" color="secondary" />,
    );
    expect(
      screen.getByRole("link", { name: "How to use" }),
    ).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveAttribute("href", "/howto");
    expect(container).toMatchSnapshot();
  });

  test("デフォルトカラーのテスト", () => {
    render(<Button title="テスト" link="/test" />);
    expect(screen.getByRole("link")).toHaveClass("bg-primary");
  });
});
