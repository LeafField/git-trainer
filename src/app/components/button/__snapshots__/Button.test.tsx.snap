// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttonのレンダリングテスト primaryのレンダリングテスト 1`] = `
<div>
  <a
    class="block w-fit rounded-[2.8125rem] px-5 py-2 text-base font-normal text-white after:block after:h-[1px]
      after:w-full after:scale-0 after:transition-all hover:after:scale-100 hover:after:bg-white
      hover:after:content-[""] md:px-10 md:py-4 md:text-2xl
      bg-primary false"
    href="/difficulty"
  >
    Let's Play!
  </a>
</div>
`;

exports[`Buttonのレンダリングテスト secondaryのレンダリングテスト 1`] = `
<div>
  <a
    class="block w-fit rounded-[2.8125rem] px-5 py-2 text-base font-normal text-white after:block after:h-[1px]
      after:w-full after:scale-0 after:transition-all hover:after:scale-100 hover:after:bg-white
      hover:after:content-[""] md:px-10 md:py-4 md:text-2xl
      false border border-white"
    href="/howto"
  >
    How to use
  </a>
</div>
`;
